data/abi/abi.go

```diff
@@ -0,0 +1,1164 @@
+		ABIType: typeUint,
+		value:   value,
+	}, nil
+}
+
+// MakeUfixed takes a big rational number representation, a type size, and a type precision,
+// and returns an ABI Value of ABI UFixed<size>x<precision>
+func MakeUfixed(value *big.Int, size uint16, precision uint16) (Value, error) {
+	ufixedValueType, err := MakeUfixedType(size, precision)
+	if err != nil {
+		return Value{}, err
+	}
+	uintVal, err := MakeUint(value, size)
+	if err != nil {
+		return Value{}, err
+	}
+	uintVal.ABIType = ufixedValueType
+	return uintVal, nil
+}
+
+// MakeString takes a string and returns an ABI String type Value.
+func MakeString(value string) Value {
+	return Value{
+		ABIType: MakeStringType(),
+		value:   value,
+	}
+}
+
+// MakeByte takes a byte and returns an ABI Byte type value.
+func MakeByte(value byte) Value {
+	return Value{
+		ABIType: MakeByteType(),
+		value:   value,
+	}
+}
+
+// MakeAddress takes an [32]byte array and returns an ABI Address type value.
+func MakeAddress(value [32]byte) Value {
+	return Value{
+		ABIType: MakeAddressType(),
+		value:   value,
+	}
+}
+
+// MakeDynamicArray takes an array of ABI value (can be empty) and element type,
+// returns an ABI dynamic length array value.
+func MakeDynamicArray(values []Value, elemType Type) (Value, error) {
+	if len(values) >= (1 << 16) {
+		return Value{}, fmt.Errorf("dynamic array make error: pass in argument number larger than 2^16")
+	}
+	for i := 0; i < len(values); i++ {
+		if !values[i].ABIType.Equal(elemType) {
+			return Value{}, fmt.Errorf("type mismatch: %s and %s",
+				values[i].ABIType.String(), elemType.String())
+		}
+	}
+	return Value{
+		ABIType: MakeDynamicArrayType(elemType),
+		value:   values,
+	}, nil
+}
+
+// MakeStaticArray takes an array of ABI value and returns an ABI static length array value.
+func MakeStaticArray(values []Value) (Value, error) {
+	if len(values) >= (1 << 16) {
+		return Value{}, fmt.Errorf("static array make error: pass in argument number larger than 2^16")
+	} else if len(values) == 0 {
+		return Value{}, fmt.Errorf("static array make error: 0 argument passed in")
+	}
+	for i := 0; i < len(values); i++ {
+		if !values[i].ABIType.Equal(values[0].ABIType) {
+			return Value{}, fmt.Errorf("type mismatch: %s and %s",
+				values[i].ABIType.String(), values[0].ABIType.String())
+		}
+	}
+	return Value{
+		ABIType: MakeStaticArrayType(values[0].ABIType, uint16(len(values))),
+		value:   values,
+	}, nil
+}
+
+// MakeTuple takes an array of ABI values and returns an ABI tuple value.
+func MakeTuple(values []Value) (Value, error) {
+	if len(values) >= (1 << 16) {
+		return Value{}, fmt.Errorf("tuple make error: pass in argument number larger than 2^16")
+	}
+	tupleType := make([]Type, len(values))
+	for i := 0; i < len(values); i++ {
+		tupleType[i] = values[i].ABIType
+	}
+
+	castedTupleType, err := MakeTupleType(tupleType)
+	if err != nil {
+		return Value{}, err
+	}
+
+	return Value{
+		ABIType: castedTupleType,
+		value:   values,
+	}, nil
+}
```

Comment:
Changed accordingly on `Make` for `static array`, `dynamic array` and `tuple`.

I think by ABI spec, `dynamic array` and `tuple` can be empty; while `static array` must have at least 1 element:
- `dynamic array` may need additional type argument, or element type cannot be inferred for empty array.
- `tuple` just need an array of value argument.